modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredEncoder.scala

```diff
@@ -0,0 +1,44 @@
+package io.circe.derivation
+
+import scala.deriving.Mirror
+import scala.compiletime.constValue
+import io.circe.{Encoder, Json, JsonObject}
+
+trait ConfiguredEncoder[A](using conf: Configuration) extends Encoder.AsObject[A], DerivedInstance[A]:
+  def elemEncoders: Array[Encoder[?]]
+  
+  final def encodeElemAt(index: Int, elem: Any, transformName: String => String): (String, Json) =
+    (transformName(elemLabels(index)), elemEncoders(index).asInstanceOf[Encoder[Any]].apply(elem))
+  
+  final def encodeProduct(a: A): JsonObject =
+    val product = a.asInstanceOf[Product]
+    val iterable = Iterable.tabulate(product.productArity) { index =>
+      encodeElemAt(index, product.productElement(index), conf.transformMemberNames)
+    }
+    JsonObject.fromIterable(iterable)
+  
+  final def encodeSum(index: Int, a: A): JsonObject =
+    val (constructorName, json) = encodeElemAt(index, a, conf.transformConstructorNames)
+    conf.discriminator match
+      case None => JsonObject.singleton(constructorName, json)
+      case Some(discriminator) => json.asObject.getOrElse(JsonObject.empty).add(discriminator, Json.fromString(constructorName))
+
+object ConfiguredEncoder:
+  inline final def derived[A](using conf: Configuration = Configuration.default)(using mirror: Mirror.Of[A]): ConfiguredEncoder[A] =
```

Comment:
Using default configuration here is error prone in my opinion: if you forget to import your own configuration or remove it by accident your code will compile anyway.

Reply:
Now Configured* derived methods no longer have a default configuration. And Encoder.AsObject, Decoder and Codec.AsObject now use the default configuration.